src/analytics/SegmentAnalyticsService.js

```diff
@@ -187,9 +187,13 @@ class SegmentAnalyticsService {
         resolve();
       });
 
-      // if the segment domain is blocked on user's network/browser, this promis does not get
+      // if the segment domain is blocked on user's network/browser, this promise does not get
       // resolved and user see a blank page, set timeout out to resolve the promise.
-      setTimeout(resolve, 2000);
+      setTimeout(() => {
+        // Other segment calls won't work if we are not identified. So let's just un-initialize ourselves.
+        this.segmentInitialized = false;
+        resolve();
+      }, 2000);
```

Comment:
How bad does it screw with our metrics if we don't have segment events for people that timeout at two seconds? How many people do we think hit this? We saw 11.6k occurances of the segment error in 5 days. Which I guess isn't enormous. Does that number feel right?

Reply:
I'm not a fan of using arbitrary timeouts like this, honestly... I didn't get a chance to chime in on the previous PR that added this timeout.  On poor connections, 2 seconds may be reasonable; seems like we're likely to drastically skew our metrics if we start excluding users with the slowest connections.

Reply:
We're hitting this, presumably, if `global.analytics.ready` doesn't fire.  I'd like to understand the conditions under which it doesn't fire and see if we can't detect those, rather than just delaying.